coreos-koji-tagger/coreos_koji_tagger.py

```diff
@@ -305,15 +305,18 @@ def process_lockfiles(self, commit=None):
         desiredrpms = set()
         for arch in ['x86_64', 'aarch64', 'ppc64le', 's390x']:
             for lockfile in ['manifest-lock', 'manifest-lock.overrides']:
-                url = f'https://raw.githubusercontent.com/{self.github_repo_fullname}/{rev}/{lockfile}.{arch}.json'
-                logger.info(f'Attempting to retrieve data from {url}')
-                r = requests.get(url)
-                if r.ok:
-                    # parse the lockfile and add the set of rpm NEVRAs (strings)
-                    desiredrpms.update(parse_lockfile_data(r.text))
-                else:
-                    # Log any errors we encounter. 404s are ok, but won't hurt to log
-                    logger.warning('URL request error: %s' % r.text.strip())
+                for filetype in ['yaml', 'json']:
+                    url = f'https://raw.githubusercontent.com/{self.github_repo_fullname}/{rev}/{lockfile}.{arch}.{filetype}'
+                    logger.info(f'Attempting to retrieve data from {url}')
+                    r = requests.get(url)
+                    if r.ok:
+                        # parse the lockfile and add the set of rpm NEVRAs (strings)
+                        desiredrpms.update(parse_lockfile_data(r.text, filetype))
+                        break # If both yaml and json files exist, only parse one 
+                              # of them. Prefer yaml.
+                    else:
+                        # Log any errors we encounter. 404s are ok, but won't hurt to log
+                        logger.warning('URL request error: %s' % r.text.strip())
```

Comment:
Not directly related, though can we do a check on whether `desiredrpms` is empty here and return early if so?

Reply:
IIUC the only reason that `desiredrpms` would be empty is if no lockfiles existed in the repo at all. Does that match your understanding?

If so, is targeting a repo with no lockfiles valid. If valid is it something we want to optimize for?

Reply:
I was thinking more about it in terms of reasoning through the code than optimizing for it. Getting the trivial cases out of the way means we don't have to think about them when hacking on code lower down.

Reply:
ok. I'll address in a followup: https://github.com/coreos/fedora-coreos-releng-automation/pull/62

Reply:
Sound good?